android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsViewModel.kt

```diff
@@ -52,4 +59,60 @@ class AncDetailsViewModel(
     }
     return patientCarePlan
   }
+
+  fun fetchCQLLibraryData(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource
+  ): LiveData<String> {
+    var libraryData = MutableLiveData<String>()
+    viewModelScope.launch(dispatcher.io()) {
+      val auxCQLLibraryData =
+        parser.encodeResourceToString(
+          fhirResourceDataSource.loadData(LIBRARY_URL).entry[0].resource
+        )
+      libraryData.postValue(auxCQLLibraryData)
+    }
+    return libraryData
+  }
+
+  fun fetchCQLFhirHelperData(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource
+  ): LiveData<String> {
+    var helperData = MutableLiveData<String>()
+    viewModelScope.launch(dispatcher.io()) {
+      val auxCQLHelperData =
+        parser.encodeResourceToString(fhirResourceDataSource.loadData(HELPER_URL).entry[0].resource)
+      helperData.postValue(auxCQLHelperData)
+    }
+    return helperData
+  }
+
+  fun fetchCQLValueSetData(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource
+  ): LiveData<String> {
+    var valueSetData = MutableLiveData<String>()
+    viewModelScope.launch(dispatcher.io()) {
+      val auxCQLValueSetData =
+        parser.encodeResourceToString(fhirResourceDataSource.loadData(VALUE_SET_URL))
+      valueSetData.postValue(auxCQLValueSetData)
+    }
+    return valueSetData
+  }
+
+  fun fetchCQLPatientData(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource,
+    patientId: String
+  ): LiveData<String> {
+    var patientData = MutableLiveData<String>()
+    PATIENT_URL = "https://fhir.labs.smartregister.org/fhir/Patient/$patientId/\$everything"
```

Comment:
I think the base URL for this also belongs in a config file

Reply:
@pld This is done